db/package.json

```diff
@@ -1,10 +1,11 @@
 {
   "name": "deploy",
   "dependencies": {
-    "@sap/hdi-deploy": "^3"
+    "@sap/hdi-deploy": "^3",
+    "@sap/cds-dk": "^3"
```

Comment:
Hm, that's not really nice that we now need a `@sap/cds-dk` dependency here. Would a `devDependency` also be sufficient? This would be the better solution I guess, if required at all.

Reply:
I also don't like it but we need the the `cds` command here to build the module. The problem is that the mbt build tool is installing the npm modules with "--production", so a dev dependency would not be considered.

Reply:
@chgeo what are your opinions on this?

Reply:
Don't like this approach here either.  We do have a solution for this in the mta that we generate for new Java projects.
I vote for not inventing new build mechanisms here, but just use what we offer to new projects.

```
cds add javabuildtest --add java,mta,hana,samples
```
gives you in `mta.yaml`:
```yaml
build-parameters:
  before-all:
   - builder: custom
     commands:
      - npm install --production
      - npx -p @sap/cds-dk cds build --production
```
No dependencies in `package.json`, `db/package.json` and no postinstall scripts.

Reply:
Fully aggree that this should be consistent. We did this change https://github.com/SAP-samples/cloud-cap-samples-java/pull/40 as part of an experiment we discussed with Lothar, but if it creates these follow-up problems, we should probably rather revert back to the previous approach / the one we promote currently with  `cds init`.

Reply:
The current approach promoted by `cds init` has the drawback that the cds build is executed twice in case of an MTAR build.

Reply:
Only the java part should be executed twice.

Reply:
Right, this could be improved.  But rather should do that w/ Lothar and Markus instead of having yet another custom solution to the problem.
Double build time is not an issue here.

Reply:
Yes fully agree. @MarcelMerkle can you revert the changes and implement the solution based on the way `cds init` currently behaves. That the Java module only builds the srv part can be kept as a continued experiment I guess. If we like how that feels we can also update that in our Maven archetype.

Reply:
I really like `npx -p @sap/cds-dk cds build --production`! Didn't know that this was possible.

Reply:
To optimize the build time, we could think of specifying only the necessary build tasks for the global cds-build. But for now I would keep it consistent with `cds init`.

Reply:
@chgeo What is the reason for the `npm install --production` step? The `package.json` in the root directory has no dependencies.